tests/pdb_test.go

```diff
@@ -0,0 +1,115 @@
+	"kubevirt.io/kubevirt/tests/exec"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libpod"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[sig-compute]Pod Disruption Budget (PDB)", decorators.SigCompute, func() {
+	var (
+		vmi *v1.VirtualMachineInstance
+		err error
+		pdb *policyv1.PodDisruptionBudget
+	)
+
+	BeforeEach(func() {
+		vmi = libvmifact.NewCirros(libvmi.WithLogSerialConsole(true))
+		vmi.Spec.Domain.Devices.AutoattachSerialConsole = pointer.Bool(true)
+		vmi.ObjectMeta.Labels = map[string]string{}
+		vmi.ObjectMeta.Labels["kubevirt.io/vmi"] = vmi.Name
+		vmi.Spec.EvictionStrategy = kvpointer.P(v1.EvictionStrategyLiveMigrate)
+
+		By("Creating a VMI")
+		vmi, err = kubevirt.Client().VirtualMachineInstance(testsuite.NamespaceTestDefault).Create(context.Background(), vmi, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Waiting for VMI to be created")
+		Eventually(func() (*v1.VirtualMachineInstance, error) {
+			return kubevirt.Client().VirtualMachineInstance(testsuite.NamespaceTestDefault).Get(context.Background(), vmi.Name, metav1.GetOptions{})
+		}, 60*time.Second, 1*time.Second).Should(BeAssignableToTypeOf(&v1.VirtualMachineInstance{}))
+
+		By("Creating a PodDisruptionBudget for the VMI")
+		pdb = &policyv1.PodDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pdb-for-vmi",
+				Namespace: testsuite.NamespaceTestDefault,
+				OwnerReferences: []metav1.OwnerReference{
+					*metav1.NewControllerRef(vmi, v1.VirtualMachineInstanceGroupVersionKind),
+				},
+			},
+			Spec: policyv1.PodDisruptionBudgetSpec{
+				MinAvailable: &intstr.IntOrString{IntVal: 1},
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{"kubevirt.io/vmi": vmi.Name},
+				},
+			},
+		}
+		_, err = kubevirt.Client().PolicyV1().PodDisruptionBudgets(testsuite.NamespaceTestDefault).Create(context.Background(), pdb, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if CurrentSpecReport().Failed() {
+			if vmi != nil {
+				virtlauncherPod, err := libpod.GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
+				if err == nil {
+					artifactsDir, _ := os.LookupEnv("ARTIFACTS")
+					outputString, err := exec.ExecuteCommandOnPod(virtlauncherPod, "guest-console-log", []string{"/bin/bash", "-c", "/bin/tail -v -n +1 " + fmt.Sprintf("/var/run/kubevirt-private/%v/virt-serial*-log*", vmi.UID)})
+					if err == nil {
+						lpath := filepath.Join(artifactsDir, fmt.Sprintf("serial_logs_content_%v.txt", vmi.UID))
+						_, _ = fmt.Fprintf(GinkgoWriter, "Serial console log failed, serial console logs dump from virt-launcher pod collected at file at %s\n", lpath)
+						_ = os.WriteFile(lpath, []byte(outputString), 0644)
+					}
+				}
+			}
+		}
+	})
+
+	It("should ensure the Pod Disruption Budget (PDB) is deleted when the VM is stopped via guest OS", func() {
```

Comment:
If this is a single test I'd suggest you move everything (including the test initialization code) inside the `It` fixture, having it split between a BeforeEach() and an It() makes sense only if you have more than one test.

Otherwise it's slightly less readable because you have the test split between 2 places.

Reply:
agreed